### PR TITLE
DAOS-9690 test: Add support to use label in tests for daos container …

### PR DESCRIPTION
**Pull request by shimizukko**
…create/destroy

The goal is to convert the tests to use label to create and destroy
container. This makes it a complex PR, so I broke it down to 3 parts.
This is the first part that does the following:

1. Move LabelGenerator from test_utils_pool.py to test_utils_base.py
because both TestPool and TestContainer use it.
2. Add self.label in daos_utils_base.py -> class CreateSubCommand
to add --label support.
3. Add label parameter in daos_utils.py -> container_create() and
pass it in to _get_json_result()

Skip-unit-tests: true
Test-tag: pr
Signed-off-by: Makito Kano <makito.kano@intel.com>

### CODE REVIEW

File: src/tests/ftest/util/test_utils_base.py [hunk 1]
```diff
@@ -183,3 +183,32 @@ def _check_info(self, check_list):
                 self.log.error(msg)
                 check_status = False
         return check_status
+
+
+class LabelGenerator():
+    # pylint: disable=too-few-public-methods
+    """Generates label used for pool."""
```
Comment (daltonbohning) [inline src/tests/ftest/util/test_utils_base.py:190]: Minor
```suggestion
    """Generates labels used for pools and containers."""
```

**Review by daltonbohning — APPROVED** [review]

File: src/tests/ftest/util/test_utils_base.py (hunk 1 above)
Comment (shimizukko) [reply src/tests/ftest/util/test_utils_base.py:190]: Thanks. I'll fix it in the upcoming PR.

**Review by phender — CHANGES_REQUESTED** [review]
The `LabelGenerator` class is imported by 9 other non-skipped tests that were not run as part of this PR:

- nvme/pool_exclude.py:NvmePoolExclude.test_nvme_pool_excluded (weekly)
- osa/offline_extend.py:OSAOfflineExtend.test_osa_offline_extend (daily)
- osa/offline_extend.py:OSAOfflineExtend.test_osa_offline_extend_without_checksum (weekly)
- osa/offline_extend.py:OSAOfflineExtend.test_osa_offline_extend_multiple_pools (weekly)
- osa/offline_extend.py:OSAOfflineExtend.test_osa_offline_extend_after_snapshot (daily)
- osa/offline_parallel_test.py:OSAOfflineParallelTest.test_osa_offline_parallel_test (daily)
-osa/offline_parallel_test.py:OSAOfflineParallelTest.test_osa_offline_parallel_test_without_csum (weekly)
- osa/offline_parallel_test.py:OSAOfflineParallelTest.test_osa_offline_parallel_test_rank_boot (weekly)
- osa/offline_parallel_test.py:OSAOfflineParallelTest.test_osa_offline_parallel_test_oclass (weekly)

The import statement used by these tests may need to updated:
```
from test_utils_pool import TestPool, LabelGenerator
```
But at the very least these tests should be run as part of the PR to ensure that changes have no ill effects.

Note: the osa/online_reintegration.py test also uses this import, but all tests are currently skipped.

**Comment by daltonbohning** [comment]
@phender python actually allows you to import from a file that imports from a file. So `from test_utils_pool import LabelGenerator` is valid, because `test_utils_pool` imports `LabelGenerator`. Otherwise, it seems the `LabelGenerator` class was just copy-pasted, with no real changes.
That being said, looking at this again... I do think it would be cleaner and safer to update these imports and run the tests just in case.

**Comment by shimizukko** [comment]
@phender Thanks. I updated the import statements and ran those tests that directly import `LabelGenerator`.

**Review by daltonbohning — APPROVED** [review]

**Review by phender — APPROVED** [review]

**Pull request merged**